torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -125,9 +125,6 @@ std::shared_ptr<FutureMessage> sendMessageWithAutograd(
 
   auto fut = agent.send(dst, std::move(msg));
   if (rf != nullptr) {
-    // save the local threadId so that end() callbacks can be correctly invoked
-    // from a different thread.
-    rf->setThreadId();
```

Comment:
Nice, this will definitely make the API a lot cleaner for profiling async use cases